rcsb/workflow/wuw/PdbCsmImageWorkflow.py

```diff
@@ -0,0 +1,99 @@
+logger = logging.getLogger()
+
+
+class PdbCsmImageWorkflow:
+
+    def imagesGenJpgs(self, **kwargs: dict) -> None:
+        """Generate jpgs for given pdb/csm list."""
+        idListName = kwargs.get("idListName", None)
+        idListFile = os.path.join(kwargs.get("idListPath"), idListName)
+
+        jpgXvfbExecutable = kwargs.get("jpgXvfbExecutable")
+        jpgScreen = kwargs.get('jpgScreen')
+        molrenderExe = kwargs.get("molrenderExe")
+        jpgRender = kwargs.get("jpgRender")
+        jpgHeight = str(kwargs.get("jpgHeight"))
+        jpgWidth = str(kwargs.get("jpgWidth"))
+        jpgFormat = kwargs.get("jpgFormat")
+        jpgAdditionalCmds = kwargs.get("jpgAdditionalCmds", None)
+
+        logger.info('using id file %s', idListFile)
+
+        fileObj = Path(idListFile)
+        if not (fileObj.is_file() and fileObj.stat().st_size > 0):
+            logger.error('Missing idList file %s', idListFile)
+            return
+
+        mU = MarshalUtil()
+        idList = mU.doImport("idListFile", fmt="list")
+        if not isinstance(idList, list):
+            raise TypeError("idList not a list")
+
+        for line in idList:
+
+            fileId, bcifFileName, sdm = line.split()
+            contentTypeDir = "pdb" if sdm == "experimental" else "csm"
+            logger.info('Running %s %s %s', fileId, bcifFileName, sdm)
+
+            bcifFilePath = os.path.join(kwargs.get("pdbBaseDir"), bcifFileName) if sdm == "experimental" else os.path.join(kwargs.get("csmBaseDir"), bcifFileName)
+
+            outPath = os.path.join(kwargs.get("jpgsOutDir"), contentTypeDir)
+            Path(outPath).mkdir(parents=True, exist_ok=True)
+
+            # runMolrender
+            cmd = [
+                jpgXvfbExecutable,
+                "-a",
+                "-s", f"-ac -screen 0 {jpgScreen}",
+                molrenderExe,
+                jpgRender,
+                bcifFilePath,
+                outPath,
+                "--height", jpgHeight,
+                "--width", jpgWidth,
+                "--format", jpgFormat,
+            ]
+            if jpgAdditionalCmds:
+                cmd = [*cmd, jpgAdditionalCmds]
+
+            fileObj = Path(bcifFilePath)
+            if fileObj.is_file() and fileObj.stat().st_size > 0:
+                # logger.info('Running %s', ' '.join(cmd))
+                try:
+                    result = subprocess.run(cmd, capture_output=True, text=True, check=True)
+                    logger.info(result.stdout)
+                except subprocess.CalledProcessError as e:
+                    logger.error("Error: %s", e)
+                    logger.error("Stderr: %s", e.stderr)
+
+                # check result
+                outJpgFile = os.path.join(outPath, fileId + "_model-1.jpeg")
+                fileObj = Path(outJpgFile)
+                if not (fileObj.is_file() and fileObj.stat().st_size > 0):
+                    logger.error("No image file: %s.", outJpgFile)
```

Comment:
This seems to be a serious enough error that requires an exception thrown so that the workflow knows about the problem.